distributed/src/main/java/com/orientechnologies/orient/server/distributed/impl/ODistributedDatabaseImpl.java

```diff
@@ -651,6 +654,9 @@ public void run() {
             checkTxTimeout();
           }
         };
+    final long timeout = OGlobalConfiguration.DISTRIBUTED_TX_EXPIRE_TIMEOUT.getValueAsLong();
+    timer = new Timer("Distributed database tx expiry", true);
+    timer.scheduleAtFixedRate(txTimeoutTask, timeout / 3, timeout / 3);
```

Comment:
We do have actually a global timer in the context so ` context.schedule` should do, just to keep on check the number of threads running.

Probably is also better to execute the operation in the context thread pool, so do ~ ` context.execute(() -> checkTxTimeout())` because the timer is shared is better that the timer do just timing, I can add this myself later

(and did we really missed that? :facepalm: )